pkg/controller/authconfig/authconfig.go

```diff
@@ -0,0 +1,299 @@
+	authenticationConfigControllerWorkQueueKey = "key"
+)
+
+type AuthenticationConfigController struct {
+	operatorClient *api.OperatorClient
+	authentication configv1client.AuthenticationInterface
+	services       corev1client.ServicesGetter
+	endpoints      corev1client.EndpointsGetter
+	route          routeclient.RouteInterface
+	queue          workqueue.RateLimitingInterface
+}
+
+func NewAuthenticationConfigController(
+	configClient configclient.Interface,
+	kubeClient kubernetes.Interface,
+	routeClient routeclient.RouteV1Interface,
+	operatorClient *api.OperatorClient) *AuthenticationConfigController {
+	return &AuthenticationConfigController{
+		operatorClient: operatorClient,
+		authentication: configClient.ConfigV1().Authentications(),
+		route:          routeClient.Routes("openshift-authentication"),
+		services:       kubeClient.CoreV1(),
+		endpoints:      kubeClient.CoreV1(),
+		queue:          workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "IngressStateController"),
+	}
+}
+
+func (c *AuthenticationConfigController) Run(workers int, stopCh <-chan struct{}) {
+	defer utilruntime.HandleCrash()
+	defer c.queue.ShutDown()
+
+	klog.Infof("Starting AuthenticationConfigController")
+	defer klog.Infof("Shutting down AuthenticationConfigController")
+
+	// doesn't matter what workers say, only start one.
+	go wait.Until(c.runWorker, time.Second, stopCh)
+
+	// add time based trigger
+	go wait.Until(func() { c.queue.Add(authenticationConfigControllerWorkQueueKey) }, time.Minute, stopCh)
+
+	<-stopCh
+}
+
+func (c *AuthenticationConfigController) runWorker() {
+	for c.processNextWorkItem() {
+	}
+}
+
+func (c *AuthenticationConfigController) processNextWorkItem() bool {
+	dsKey, quit := c.queue.Get()
+	if quit {
+		return false
+	}
+	defer c.queue.Done(dsKey)
+
+	err := c.sync()
+	if err == nil {
+		c.queue.Forget(dsKey)
+		return true
+	}
+
+	utilruntime.HandleError(fmt.Errorf("%v failed with : %v", dsKey, err))
+	c.queue.AddRateLimited(dsKey)
+
+	return true
+}
+
+func (c *AuthenticationConfigController) sync() error {
+	authConfigNoDefaults, err := c.authentication.Get("cluster", metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	_, operatorStatus, _, err := c.operatorClient.GetOperatorState()
+	if err != nil {
+		return err
+	}
+
+	route, err := c.route.Get("oauth-openshift", metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	expectedReference := configv1.ConfigMapNameReference{
+		Name: "oauth-openshift",
+	}
+
+	if authConfigNoDefaults.Status.IntegratedOAuthMetadata == expectedReference {
+		return nil
+	}
+
+	authConfigNoDefaults.Status.IntegratedOAuthMetadata = expectedReference
+	_, err = c.authentication.UpdateStatus(authConfigNoDefaults)
+	if err != nil {
+		return err
+	}
+
+	wellknownReady, wellknownMsg, err := c.checkWellknownEndpointsReady(authConfigNoDefaults, route)
+	prefix := "WellKnownEndpoint"
+	if err != nil {
+		v1helpers.SetOperatorCondition(&operatorStatus.Conditions,
```

Comment:
This does not do anything, the original logic was working differently. Use `v1helpers.UpdateStatus`.

Reply:
updated it as per our slack discussion!